### PR TITLE
openblas: 0.3.10 -> 0.3.12

### LINKED ISSUE #92458: python36Packages.scipy: segmentation fault when running tests
**Describe the bug**
On current nixos-unstable Scipy fails to build with a segmentation fault during testing:

```
../cluster/tests/test_vq.py::TestKMean::test_kmeans2_init <- ../../nix/store/d00fds078va595ifakggj6srmky01jq2-python3.6-scipy-1.4.1/lib/python3.6/site-packages/scipy/cluster/tests/test_vq.py PASSED [  0%]
../cluster/tests/test_vq.py::TestKMean::test_krandinit <- ../../nix/store/d00fds078va595ifakggj6srmky01jq2-python3.6-scipy-1.4.1/lib/python3.6/site-packages/scipy/cluster/tests/test_vq.py Fatal Python error: Segmentation fault

Current thread 0x00007ffff7898f40 (most recent call first):
  File "/nix/store/d00fds078va595ifakggj6srmk/nix/store/ds7r5c4z1vzrm0jw7ay712zs59p64hf7-stdenv-linux/setup: line 1302:  9640 Segmentation fault      (core dumped) /nix/store/acwrn1di4lgiz80qxam9vk28b2mkr2py-python3-3.6.10/bin/python3.6 -c 'import scipy; scipy.test("fast", verbose=10)'
builder for '/nix/store/2sa0mm8g3rb2l4x4j4y2i7qihb2lbdpy-python3.6-scipy-1.4.1.drv' failed with exit code 139
cannot build derivation '/nix/store/0mxrinnvr4f43h17wpcfpfisc3kbqywk-python3-3.6.10-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0mxrinnvr4f43h17wpcfpfisc3kbqywk-python3-3.6.10-env.drv' failed
```

**To Reproduce**
1. save the following as `shell.nix`
```nix
{
  nixpkgs ?  import (builtins.fetchGit {
    url = "https://github.com/nixos/nixpkgs-channels/";

    #
    # Commit hash for nixos-unstable
    # `git ls-remote https://github.com/nixos/nixpkgs-channels nixos-unstable`
    name = "nixos-unstable-2020-07-06";
    ref = "refs/heads/nixos-unstable";
    rev = "dc80d7bc4a244120b3d766746c41c0d9c5f81dfa";

    #
    # Commit hash for nixos-20.03
    # `git ls-remote https://github.com/nixos/nixpkgs-channels nixos-20.03`
    #name = "nixos-20.03-2020-07-06";
    #ref = "refs/heads/nixos-20.03";
    #rev = "afa9ca61924f05aacfe495a7ad0fd84709d236cc";
  }) {}
}:

with nixpkgs;

let
  python = python36;
  python-with-base-packages = python.withPackages(ps: with ps; [
    # the following packages are related to the dependencies of your python
    # project.
    # In this particular example the python modules listed in the
    # requirements.txt require the following packages to be installed locally
    # in order to compile any binary extensions they may require.
    #

    scipy
  ]);

in mkShell {

  buildInputs = [
    python-with-base-packages
    python3Packages.venvShellHook
  ];

  shellHook = ''
    # set SOURCE_DATE_EPOCH so that we can use python wheels
    SOURCE_DATE_EPOCH=$(date +%s)
    venvShellHook
  '';
}

```
2. run `nix-shell`

**Expected behavior**

Should not segfault during test.

**Additional context**
SciPY seems to build just fine on current 20.03 release.

**Notify maintainers**
cc @FRidh , @matthewbauer 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.46, NixOS, 20.09pre232692.7db146538e4 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"home-manager, nixos-20.09pre233323.dc80d7bc4a2, nixos-hardware, nixpkgs-20.09pre233075.6e139365424"`
 - channels(ente): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by glittershark**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/92458

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/math/openblas/default.nix [hunk 1]
```diff
@@ -144,12 +144,6 @@ stdenv.mkDerivation rec {
     buildPackages.stdenv.cc
   ];
 
-  # Disable an optimisation which seems to cause issues, pending an
-  # upstream fix: https://github.com/xianyi/OpenBLAS/issues/2496
```
Comment (glittershark) [inline pkgs/development/libraries/science/math/openblas/default.nix:148]: this appears to have been closed upstream

**Comment by glittershark** [comment]
cc @FRidh

**Comment by FRidh** [comment]
Please test/build numpy, scipy, and octave.

**Comment by glittershark** [comment]
- [X] numpy
- [X] scipy
- [ ] octave

**Comment by flokli** [comment]
`octave` seems to be fine too - this should however target staging due to the amount of rebuilds.

**Comment by glittershark** [comment]
😢 that wasn't what I wanted

**Comment by glittershark** [comment]
> `octave` seems to be fine too - this should however target staging due to the amount of rebuilds.

done.

**Comment by flokli** [comment]
This needs a rebase.

**Comment by glittershark** [comment]
> This needs a rebase.

done

**Comment by flokli** [comment]
Apparently this wasn't pushed, or at least GitHub still shows a conflict.

**Comment by glittershark** [comment]
for real this time

**Comment by jonringer** [comment]
this breaks numpy for me, or at least now, it segfaults when running tests with 32 cores
```
store/gvkhgh813chlsvdsnxgwhv72xns08xcp-python3-3.8.6/bin/python3.8 -c 'import numpy; numpy.test("fast", verbose=10)'
builder for '/nix/store/s7azf4dald37gw9sa9shpvvpgldnjwgz-python3.8-numpy-1.19.1.drv' failed with exit code 139
error: build of '/nix/store/s7azf4dald37gw9sa9shpvvpgldnjwgz-python3.8-numpy-1.19.1.drv' failed
3b4cd4f4dacd7db4011e88d819a1ae5acc3f3aff is the first bad commit
commit 3b4cd4f4dacd7db4011e88d819a1ae5acc3f3aff
Author: Griffin Smith <root@gws.fyi>
Date:   Mon Oct 26 13:08:56 2020 -0400

    openblas: 0.3.10 -> 0.3.12

 ...-optimised-aarch64-dgemm_beta-pending-fix.patch | 26 ----------------------
 .../libraries/science/math/openblas/default.nix    |  8 +------
 2 files changed, 1 insertion(+), 33 deletions(-)
 delete mode 100644 pkgs/development/libraries/science/math/openblas/0001-Disable-optimised-aarch64-dgemm_beta-pending-fix.patch
bisect run success
[11:22:55] jon@nixos ~/projects/nixpkgs ((3b4cd4f4dac...)|BISECTING)
$ git bisect run nix-build -A python3Packages.numpy
```
testing now if turning down the cores helps

**Comment by glittershark** [comment]
can confirm that `python3Packages.numpy` builds fine on my machine

**Comment by jonringer** [comment]
interesting, not sure why mine just segfaults

**Comment by glittershark** [comment]
I'm not sure how this all works, but can we ask Hydra to build it for us?

**Comment by jonringer** [comment]
I think ofborg was able to build this for us. Still, this shouldn't be happening :(

**Comment by andir** [comment]
This PR did not bump the version attribute. I am a bit confused how all of you did run tests but not notice the hash mismatch.

**Comment by FRidh** [comment]
Reverted in 4e29151910288bf478248d793f77d18226bf3f6d.

**Comment by xaverdh** [comment]
> This PR did not bump the version attribute. I am a bit confused how all of you did run tests but not notice the hash mismatch.

Maybe it was lost in the rebase / force push? Does github record a history of pull requests somehow?

**Comment by FRidh** [comment]
Version was there in https://github.com/NixOS/nixpkgs/commit/858e5dc235b70b6102cae2c9ddf0731522374514 before the force push.

**Comment by glittershark** [comment]
:facepalm: sorry about that - that's what I get for trying to multitask as much as I was

**Pull request merged**